python/hsfs/core/arrow_flight_client.py

```diff
@@ -397,3 +567,55 @@ def supports(self, featuregroups):
             if fg.storage_connector.type not in self.SUPPORTED_EXTERNAL_CONNECTORS:
                 return False
         return True
+
+    @property
+    def timeout(self) -> Union[int, float]:
+        """Timeout in seconds for Hopsworks Feature Query Service do_get or do_action operations, not including the healthcheck."""
+        return self._timeout
+
+    @timeout.setter
+    def timeout(self, value: Union[int, float]) -> None:
+        self._timeout = value
+
+    @property
+    def health_check_timeout(self) -> Union[int, float]:
+        """Timeout in seconds for the healthcheck operation."""
+        return self._health_check_timeout
+
+    @health_check_timeout.setter
+    def health_check_timeout(self, value: Union[int, float]) -> None:
+        self._health_check_timeout = value
+
+    @property
+    def host_url(self) -> Optional[str]:
+        """URL of Hopsworks Feature Query Service."""
+        return self._host_url
+
+    @host_url.setter
+    def host_url(self, value: str) -> None:
+        self._host_url = value
+
+    @property
+    def disabled_for_session(self) -> bool:
+        """Whether the client is disabled for the current session."""
+        return self._disabled_for_session
+
+    @property
+    def enabled_on_cluster(self) -> bool:
+        """Whether the client is enabled on the cluster."""
+        return self._enabled_on_cluster
+
+    @property
+    def missing_variable_name(self) -> Optional[str]:
+        """Name of the missing variable that caused the client to be disabled, if any."""
+        return self._missing_variable_name
+
+    @property
+    def health_check_succeeded_once(self) -> bool:
+        """Whether the healthcheck has succeeded at least once."""
+        return self._health_check_succeeded_once
```

Comment:
are these two properties necessary?

Reply:
It makes for better error messages. One is solvable only via admin intervention, the other can be temporary client or cluster issues independent of cluster config. So I would say, yes. But ready and healthcheck_succeeded_once are maybe not as valuable

Reply:
Ok no I see your point, I can merge them in one. I will do that. Thanks for the feedback